raiden-ts/src/global.d.ts

```diff
@@ -17,3 +17,9 @@ declare module 'pouchdb-adapter-indexeddb';
 declare module 'pouchdb-debug';
 
 declare type Mutable<T> = { -readonly [P in keyof T]: T[P] };
+
+declare type UnionToIntersection<U> = (U extends any ? (k: U) => void : never) extends (
+  k: infer I,
+) => void
+  ? I
+  : never;
```

Comment:
Omg I really need to catch up with some deep level TypeScript. This reads incredible. :laughing:

Reply:
Haha, this is a more or less common recipe, but this one is quite involved, it's not as flexible as I'd like, but it's useful when converting these unions to objects which actually include all properties.